app/src/main/java/com/project/myproject/ui/fragments/add_contact/AddContactsFragment.kt

```diff
@@ -0,0 +1,241 @@
+package com.project.myproject.ui.fragments.add_contact
+
+import android.Manifest
+import android.content.Context
+import android.content.pm.PackageManager
+import android.os.Build
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.app.ActivityCompat
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.fragment.app.activityViewModels
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import androidx.navigation.NavDeepLinkBuilder
+import androidx.navigation.findNavController
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.google.android.material.snackbar.Snackbar
+import com.project.myproject.R
+import com.project.myproject.data.room.entities.User
+import com.project.myproject.databinding.FragmentAddContactsBinding
+import com.project.myproject.ui.fragments.add_contact.adapter.UserAdapter
+import com.project.myproject.ui.fragments.add_contact.viewmodel.AddContactsViewModel
+import com.project.myproject.ui.fragments.base.BaseFragment
+import com.project.myproject.ui.fragments.utils.NotificationUtils
+import com.project.myproject.ui.fragments.viewmodel.SharedViewModel
+import com.project.myproject.utils.DefaultItemDecorator
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.launch
+
+@AndroidEntryPoint
+class AddContactsFragment :
+    BaseFragment<FragmentAddContactsBinding>(FragmentAddContactsBinding::inflate),
+    UserAdapter.OnUserItemCLickListener {
+
+    private val viewModel: AddContactsViewModel by viewModels()
+    private val sharedViewModel: SharedViewModel by activityViewModels()
+
+    private lateinit var adapter: UserAdapter
```

Comment:
adapter should be intialized with by lazy but I suppose I understand your choice